libr/anal/p/anal_arm_cs.c

```diff
@@ -3124,6 +3124,161 @@ static void op_fillval(RAnalOp *op , csh handle, cs_insn *insn, int bits) {
 	}
 }
 
+static int hack_handle_dp_reg(ut32 insn, RAnalOp *op) {
+	const ut8 op0 = (insn >> 30) & 0x1;
+	const ut8 op1 = (insn >> 28) & 0x1;
+	const ut8 op2 = (insn >> 21) & 0xf;
+
+	// Data-processing (2 source)
+	if (op0 == 0 && op1 == 1 && op2 == 0x6) {
+		const ut8 sf = (insn >> 31) & 0x1;
+		const ut8 S = (insn >> 29) & 0x1;
+		const ut8 opcode = (insn >> 10) & 0x1f;
+		if (sf == 1 && S == 0 && opcode == 4) {
+			// irg
+			op->type = R_ANAL_OP_TYPE_MOV;
+			return op->size = 4;
+		} else if (sf == 1 && S == 0 && opcode == 0) {
+			// subp
+			op->type = R_ANAL_OP_TYPE_SUB;
+			return op->size = 4;
+		} else if (sf == 1 && S == 0 && opcode == 5) {
+			// gmi
+			op->type = R_ANAL_OP_TYPE_MOV;
+			return op->size = 4;
+		} else if (sf == 1 && S == 1 && opcode == 0) {
```

Comment:
ddont else after return

Reply:
where?

Reply:
on every if else you have a return, the else is dead code

Reply:
I can't see it. I can't see the dead code 😵

Reply:
it's not really deadd code. but after the return there's no sense in having an else branch, because the else wont be executed. for me if i see an else, my eyes ignore the return above. it's just a syntax thing that makes it more reaable for lazy eyes